### PR TITLE
Document invoker/builder validation

### PR DESCRIPTION
**Pull request by fbiville**

### CODE REVIEW

**Comment by scothis** [comment]
or run the tests

```sh
make test
```


It seems like this is actually describing the flow when not using the builder, or developing a specific invoker. Or developing a function and you can't/don't want to use the builder. Either way the docs needs to be framed for an audience to accomplish some goal.

File: README.adoc [hunk 1]
```diff
@@ -67,3 +67,7 @@ You should get something similar to
 Successfully built f36967212b6c
 Successfully tagged your-org/my-function
 ----
+
+=== Validating a custom Builder or Invoker
+
+See https://github.com/projectriff/builder/blob/master/INVOKER_VALIDATION.md{INVOKER_VALIDATION.md} for several ways of validating a custon Builder or Invoker.
```
Comment (jldec) [inline README.adoc:73]: This think this link needs [] to render for adoc.

**Comment by jldec** [comment]
Documenting how to modify and test invokers without a builder is helpful for anyone venturing down the road of creating their own builder. Publishing this useful information here is as good a place as any other to start.

File: INVOKER_VALIDATION.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+ $ INPUT_NAMES='words,numbers' \
+    INPUT_START_OFFSETS='latest,latest' \
+    OUTPUT_NAMES='repeated' \
+    FUNCTION='localhost:8081' \
+    GROUP='somehow-irrelevant' \
+    CNB_BINDINGS='/var/folders/t5/04x3n9gx2c1b74vrljr4sn6h0000gq/T/local-bindings5922978358290298314' \
+    mvn exec:java -Dstart-class="io.projectriff.processor.Processor"
+```
+
+Finally, clone the [`stream-client-go` fork](https://github.com/fbiville/stream-client-go/tree/stream_publisher):
```
Comment (fbiville) [inline INVOKER_VALIDATION.md:121]: I'm not sure we want to go ahead with this :)

**Comment by fbiville** [comment]
> It seems like this is actually describing the flow when not using the builder, or developing a specific invoker. Or developing a function and you can't/don't want to use the builder. Either way the docs needs to be framed for an audience to accomplish some goal.

I hopefully clarified the document intent and target audience. Let me know if that's still unclear.

File: INVOKER_VALIDATION.md (hunk 2 above)
Comment (jldec) [reply INVOKER_VALIDATION.md:121]: I suggest using dev-utils in cluster to publish:
https://projectriff.io/docs/v0.6/runtimes/streaming#testing-with-dev-utils

**Comment by fbiville** [comment]
> Most of this document should move into the *-function-(buildpack|invoker) repos, as it has little to do with the builder.

I agree this repo may be not the ideal place.
However, I think there is value in documenting the process in a single document.
Do you think it's better to split it up? What about the common steps?

**Comment by scothis** [comment]
> I agree this repo may be not the ideal place.
However, I think there is value in documenting the process in a single document.
Do you think it's better to split it up?

Information about how to build and test a component belongs with that component as it should be updated as that component evolves. There may also be version drift between the invoker, buildpacks and the builder that will be hard to accommodate in this repo.

> What about the common steps?

Duplicate them everywhere they need to be followed. The only bit this seems to apply to is starting the stream processor. But that seems like a odd way to test drive an invoker/function. The point of the invoker spec is that different clients should be able to plug-in and drive the invoker.

**Pull request closed without merging**